### PR TITLE
Feat/docs

### PR DESCRIPTION
**Pull request by eduardconstantin**
### Which issue(s) does this pull-request address?

Closes: #74 

### Description

Added index.html to docs folder to render documentation using docsify.
Added Readme.md as homepage.
Added _sidebar.md to have a custom sidebar.
This are the deployment steps from docsify docs: https://docsify.js.org/#/deploy
image
image
image

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [ ] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: docs/_sidebar.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+- Modules
+
+  - [Agents](agents.md)
+  - [LLMS](llms.md)
+  - [Templates](templates.md)
+  - [Memory](memory.md)
+  - [Tools](tools.md)
+  - [Cache](cache.md)
+  - [SQL Tools](sql-tools.md)
```
Comment (Tomas2D) [inline docs/_sidebar.md:14]: I added a few more modules.

Comment (Tomas2D) [reply docs/_sidebar.md:14]: If possible, please put `SQL Tools` under `Tools`.

File: docs/_sidebar.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+- Getting started
+
+  - [Home](/)
+  - [Overview](overview.md)
+
+- Modules
+
+  - [Agents](agents.md)
+  - [LLMS](llms.md)
```
Comment (Tomas2D) [inline docs/_sidebar.md:9]: `LLMs`

**Review by Tomas2D — CHANGES_REQUESTED** [review]
Great! Could you also please set up GitHub Action?

**Comment by eduardconstantin** [comment]
@Tomas2D I added the action, I used this: https://github.com/marketplace/actions/docsify-action?version=v1.1
I also updated the sidebar
image

**Review by Tomas2D — CHANGES_REQUESTED** [review]
Great. However, there are some issues:

- It makes sense to leverage `https://github.com/docsifyjs/docsify-cli` so we could develop the docs in the watch mode (update `docs:build` script and add `docs:watch`)
- auto copy `README.md` to `docs/README.md` as part of the `docs:build`
- ensures that all code snippets are rendered correctly
- ensures that all links will point to the `https://i-am-bee.github.io/bee-agent-framework/` (if not developed locally)
- ensures that formatting [!NOTE] / [!TIP] and renders correctly because, from the provided screenshots, it does not.
- ensures that pipelines are passing (DCO / Lints)

Note: the `hacktoberfest` label is attached to the related issue; isn't that sufficient?

**Comment by eduardconstantin** [comment]
ensures that formatting [!NOTE] / [!TIP] and renders correctly because, from the provided screenshots, it does not.
image

it seems that docsify needs pulgins to format different tags

**Comment by eduardconstantin** [comment]
- [ ] It makes sense to leverage https://github.com/docsifyjs/docsify-cli so we could develop the docs in the watch mode (update docs:build script and add docs:watch) - **I added the docs:watch script, I don't know what to add to build**
- [ ] auto copy README.md to docs/README.md as part of the docs:build - **The links insed the readme file needs to be changed once the file is copied to the docs folder, but this step happened during `docsify init` command, if you call this command it will override the readme.md and index.html files (the index now has plugins added inside)**
- [x] ensures that all code snippets are rendered correctly
- [x] ensures that all links will point to the https://i-am-bee.github.io/bee-agent-framework/ (if not developed locally)
- [x] ensures that formatting [!NOTE] / [!TIP] and renders correctly because, from the provided screenshots, it does not.
- [x] ensures that pipelines are passing (DCO / Lints)

File: package.json [hunk 3]
```diff
@@ -67,6 +67,7 @@
     "infra:clean-all": "yarn _docker compose down --volumes",
     "docs:build": "embedme --source-root=. docs/**/*.md",
     "docs:check": "yarn docs:build --verify",
+    "docs:watch": "docsify serve ./docs --open",
```
Comment (Tomas2D) [inline package.json:70]: add `docsify` to dev dependencies

**Comment by Tomas2D** [comment]
>I added the docs:watch script, I don't know what to add to build

`docsify generate` ?

**Comment by eduardconstantin** [comment]
> > I added the docs:watch script, I don't know what to add to build
> 
> `docsify generate` ?

this command only generates the _sidebar.md file
image

**Review by Tomas2D — CHANGES_REQUESTED** [review]
- A link under a code snippet is not working `Source: [examples/templates/objects.ts](http://localhost:3000/#/examples/templates/objects.ts)` (other links sometimes work and sometimes not, probably based on their path).
- Code is not highlighted
- When I click on a link within navbar, the target page shows, but the scroll position does not reset (so all of a sudden I am in the middle of a page instead of at its beginning)

**Comment by eduardconstantin** [comment]
image
Also fixed the rest of the issues

**Review by Tomas2D — CHANGES_REQUESTED** [review]
You only changed paths, but those code snippets are generated dynamically and may vary from branch to branch (when versioning will be introduced).

Add this block to `window.$docsify`

```
    markdown: {
          smartypants: true,
          renderer: {
            link: function(link, renderer, text) {
              const example = link.match(/examples\/(.*)/)
              if (example) {
                link = "https://github.com/i-am-bee/bee-agent-framework/tree/main/examples/".concat(link)
              }
              return this.origin.link.apply(this, [link, renderer, text]);
            }
          }
        }
```

We will modify it in the future.

**Review by Tomas2D — CHANGES_REQUESTED** [review]
Also, please rebase/update against the main and update README accordingly. 
Then, I will merge 🚀

**Review by Tomas2D — CHANGES_REQUESTED** [review]
You forgot to remove `https://github.com/i-am-bee/bee-agent-framework/tree/main` prefixes`

**Comment by eduardconstantin** [comment]
oh, yeah, sorry

**Review by Tomas2D — CHANGES_REQUESTED** [review]
Not all links are updated.

Please update them and then squash all commits into one with the following text.

```
docs: init web documentation

Ref: i-am-bee/bee=agent-framework#74
```

**Comment by eduardconstantin** [comment]
did it worked?
I used this command
`git merge --squash feature/login`

**Review by Tomas2D — APPROVED** [review]

**Comment by Tomas2D** [comment]
No problem, I will do it by myself. Thank you for your contribution.

**Pull request merged**